src/App.tsx

```diff
@@ -2,10 +2,46 @@ import React from 'react'
 import logo from './logo.svg'
 import './App.css'
 import FileImportButton from './Components/FileImportButton'
+import TextField from '@material-ui/core/TextField'
+import xml2js from 'xml2js'
+import Attempt from './types'
 
 const App = () => {
-  function handleFile() {
-    console.log('test')
+  const [attemptHistory, setAttemptHistory] = React.useState<Attempt[]>([])
+
+  function handleFile(e: React.ChangeEvent<HTMLInputElement>) {
+    if (e?.target?.files?.[0]) {
+      const file = e.target.files[0]
+
+      const reader = new FileReader()
+      reader.onload = () => {
+        const xml = reader.result?.toString()
+
+        if (xml) {
+          xml2js.parseString(xml, { trim: true }, async function(err, result) {
+            //todo: display error nicely in UI
+            if (err) {
+              console.log(err)
+            }
+
+            await getAttemptHistory(result)
+          })
+        }
+      }
+      reader.readAsText(file)
+    }
+  }
+
+  async function getAttemptHistory(xml: any) {
+    let atmptHistory: Attempt[] = []
```

Comment:
Not a big deal, but I probably wouldn't short attempt to atmpt.

Reply:
just changed it to history